share/functions/__fish_is_zfs_feature_enabled.fish

```diff
@@ -0,0 +1,18 @@
+function __fish_is_zfs_feature_enabled -a feature target -d "Returns 0 if the given ZFS feature is available or enabled for the given full-path target (zpool or dataset), or any target if none given"
+	set -l pool (echo "$target" | cut -d '/' -f1)
+    set -l feature_name ""
+    set -l pattern "[[:space:]]"
+    if test -z "$pool"
+        set pattern "$pattern$feature$pattern"
+        set feature_name (zpool get all -H | grep "$pattern")
+    else
+        set pattern "$pool$pattern$feature$pattern"
+        set feature_name (zpool get all -H $pool | grep "$pattern")
+    end
+	if test $status -ne 0 # No such feature
+		return 1
+	end
+	set -l state (echo $feature_name | cut -f3)
```

Comment:
I don't understand your replacement code. At this point, `$feature_name` contains something like this, in tab-separated values:
```
pgsql   feature@lz4_compress    active  local
```
How does your suggestion work against this?

Reply:
Do you want the `active` or `local` part or both?

`read` takes variable names (the "-l" is "--local", like in `set -l`) and splits the line it reads from stdin into those variables, using space and tab as the delimiter if nothing else has been given (via the new "--delimiter" option in 3.0). `_` here is just a throwaway variable name (we actually have `$_`, but its overwritten after each command - it's a bit of a hack currently, but it works and we should allow it).

So

```fish
echo "pgsql"\t"feature@lz4_compress"\t"active"\t"local" | read -l _ _ state
```

writes the first "field" (`pgsql`) into `$_`, the second (`lz4_compress`) into `$_` again and the rest (because there's no variable left) into `$state` (keeping the tab between "active" and "local").

So if you want just the "active" part, you'll have to add one more `_` after "state" (`read -l _ _ state _`), if you want just the "local", add one more `_` before, and if you want both in separate variables add another proper variable name.

You might be familiar with something like this from tuple-unpacking in e.g. python or go - `result, _ := somefunction()`).

Reply:
Oh, nice construction :heart_eyes:. I'm more Ruby, but I understand the concept; nice one. I was only looking for the third column, so I go with `read -l _ _ state _`